### PR TITLE
Implement `cwindow`/`lwindow` & related commands

### LINKED ISSUE #3158: quickfix/compiler error key binding
**Is your feature request related to a problem? Please describe.**
I would like to be able to use a vim command to list/navigate compiler errors.

**Describe the solution you'd like**
Commands can be mapped to actions on VS Code's problem window.

- `:copen` workbench.actions.view.problems
- `:cn[ext]` editor.action.marker.nextInFiles
- `:cp[revious]` editor.action.marker.prevInFiles
- `:ccl[ose]` focus problems panel and close

Within the problems panel allow j/k navigation

**Describe alternatives you've considered**
I've tried setting a custom key binding but the command parser errors out.
```json
"vim.normalModeKeyBindingsNonRecursive": [
        {
            "before": [":copen"],
            "commands": ["workbench.actions.view.problems"]
        }
    ]
```
https://github.com/VSCodeVim/Vim/blob/00bb9af44e0679706630f9d7ce5a43a552a67e15/src/cmd_line/parser.ts#L57-L59

A broader solution would be to allow custom command mappings without triggering an `E492` error.

**Additional context**

[Vim Quickfix documentation](http://vimdoc.sourceforge.net/htmldoc/quickfix.html)

### PR DESCRIPTION
**Pull request by mogelbrod**
**What this PR does / why we need it**:

- Map quickfix commands (`:c`...) to markers/problems panel
- Map location list commands (`:l`...) to comments panel
- Add a wrapper to make it easy to quickly map excommands to vscode commands

Caveats / discussion topics:
1. Perhaps `:l`... should be reserved for future support for navigating global searches? Should we introduce another list command prefix for problems?
2. `cnfile`/`cpfile` and `cn`/`cp` are identical, the vscode commands either:
    - navigates within the current file (editor.action.marker.next)
    - navigates each marker for all files (editor.action.marker.nextInFiles) = _the approach I went with as it seems to best align with `cn`/`cp`_
3. `lnext[f]`/`lprev[f]` doesn't navigate across files - the vscode commands doesn't seem to support it.
4. `cwindow` and `lwindow` do not close their respective panels if there are no items in their list. This felt like a reasonable compromise to avoid complicating the implementation.
5. `cclose` and `lclose` close the primary panel - not specifically their respective panel since it's not possible.

**Which issue(s) this PR fixes**

Fixes #3158

**Special notes for your reviewer**:

Feel free to suggest alternatives to the chosen excommand <> vscode command mappings!

### CODE REVIEW

**Review by J-Fields — CHANGES_REQUESTED** [review]

File: src/vimscript/exCommandParser.ts [hunk 1]
```diff
@@ -637,6 +638,15 @@ export class NoOpCommand extends ExCommand {
   }
 }
 
+function executeVsCodeCommand(command: string, ...args: unknown[]) {
```
[the remark this answers is not in the document]
Comment (mogelbrod) [reply src/vimscript/exCommandParser.ts:641]: That was my initial thinking as well, but doesn't the command need to call `succeed()` as well? The reasoning behind having a single function handle this was to minimize the noise in all of the repeated code. I'm open for doing it though, but for clarification which signature do you suggest?

```ts
// current
[['cope', 'n'], executeVsCodeCommand('workbench.panel.markers.view.focus')],
// 1
[['cope', 'n'], succeed(new VsCodeCommand('workbench.panel.markers.view.focus'))],
// 2
[['cope', 'n'], VsCodeCommand.explicit('workbench.panel.markers.view.focus')], // .explicit() calls succeed() automatically
// 3
[['cope', 'n'], succeed(VsCodeCommand.explicit('workbench.panel.markers.view.focus'))],
// or something else?
```

[the remark this answers is not in the document]
Comment (J-Fields) [reply src/vimscript/exCommandParser.ts:641]: Oh, right - I think `succeed(new VsCodeCommand(...))` is my preference

[the remark this answers is not in the document]
Comment (mogelbrod) [reply src/vimscript/exCommandParser.ts]: Do you have time to take another look @J-Fields? Sorry for the inconvenience!

**Comment by J-Fields** [comment]
Much appreciated!

**Pull request merged**